File: Programmers/1/160586. 대충 만든 자판/대충 만든 자판.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <algorithm>

using namespace std;

vector<int> solution(vector<string> keymap, vector<string> targets) {
    vector<int> answer;
    unordered_map<char, int> map;
    
    for (const string& key : keymap)
    {
        for (int i = 0; i < key.size(); i++)
        {
            char c = key[i];
            int pressCount = i + 1;
            
            if (map.find(c) == map.end())
            {
                map[c] = pressCount;
            }
            else
            {
                map[c] = min(map[c], pressCount);
            }
        }
    }
    
    for (const string& target : targets)
    {
        int sum = 0;
        bool impossible = false;
        
        for (char c : target)
        {
            if (map.find(c) == map.end())
            {
                sum = -1;
                impossible = true;
                break;
            }
            
            sum += map[c];
        }
        
        answer.push_back(impossible ? -1 : sum);
    }
    
    return answer;
}

File: Programmers/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];
        Map<Character, Integer> map = new HashMap<>();

        for (String key : keymap) {
            for (int i = 0; i < key.length(); i++) {
                char c = key.charAt(i);
                int pressCount = i + 1;
                
                map.put(c, Math.min(map.getOrDefault(c, Integer.MAX_VALUE), pressCount));
            }
        }

        for (int i = 0; i < targets.length; i++) {
            int sum = 0;
            
            for (char c : targets[i].toCharArray()) {
                if (!map.containsKey(c)) {
                    sum = -1;
                    break;
                }
                
                sum += map.get(c);
            }
            
            answer[i] = sum;
        }
        
        
        return answer;
    }
}